### PR TITLE
Add secret description and integration into springboot application

### LINKED ISSUE #36: Include a Secrets Example Lab 8
We should try to include an example using secrets in Lab https://github.com/appuio/techlab/blob/lab-3.9/labs/08_database.md similar to what we did in the kubernetes techlab.

An additional lab might then be, working with different kinds of secrets.

### PR DESCRIPTION
**Pull request by phil-pona**
Updates the Lab 8 and describes how Secrets can be used and integrated into both pods, mysql and springboot.

fixes #36

### CODE REVIEW

File: labs/08_database.md [hunk 1]
```diff
@@ -34,6 +34,91 @@ In der Web Console kann der MySQL (Ephemeral) Service via Catalog dem Projekt hi
 
 ![MySQLService](../images/lab_8_mysql.png)
 
+### Passwort und Username als plaintext?
```
Comment (bliemli) [inline labs/08_database.md]:
```suggestion
### Passwort und Username als Plaintext?
```

File: labs/08_database.md [hunk 2]
```diff
@@ -34,6 +34,91 @@ In der Web Console kann der MySQL (Ephemeral) Service via Catalog dem Projekt hi
...
+
+Beim Deployen der Datebank sowohl via CLI aber auch via Web Console, haben wir mittels Parameter oder direkt im Eingabefeld Werte für User, Passwort und Datenbank angegeben. In diesem Kapitel wollen wir uns nun anschauen, wo diese sensitiven Daten nun effektiv gelandet sind.
```
Comment (bliemli) [inline labs/08_database.md]:
```suggestion
Beim Deployen der Datebank via CLI wie auch via Web Console haben wir mittels Parameter Werte für User, Passwort und Datenbank angegeben. In diesem Kapitel wollen wir uns nun anschauen, wo diese sensitiven Daten effektiv gelandet sind.
```

File: labs/08_database.md [hunk 3]
```diff
@@ -34,6 +34,91 @@ In der Web Console kann der MySQL (Ephemeral) Service via Catalog dem Projekt hi
...
+              key: database-root-password
+              name: mysql
+        - name: MYSQL_DATABASE
+          valueFrom:
+            secretKeyRef:
+              key: database-name
+              name: mysql
+```
+
+Die Werte für die einzelnen Umgebungsvariablen kommen also aus einem sogenannten Secret, in unserem Fall hier aus dem Secret mit dem Namen `mysql`. In diesem Secret sind die vier Werte entsprechend unter den passenden Keys (`database-user`, `database-password`, `database-root-password`, `database-name`) abgelegt und können ebenso darauf zu gegriffen werden.
```
Comment (bliemli) [inline labs/08_database.md]:
```suggestion
Die Werte für die einzelnen Umgebungsvariablen kommen also aus einem sogenannten Secret, in unserem Fall hier aus dem Secret mit Namen `mysql`. In diesem Secret sind die vier Werte entsprechend unter den passenden Keys (`database-user`, `database-password`, `database-root-password`, `database-name`) abgelegt und können so referenziert werden.
```

File: labs/08_database.md [hunk 4]
```diff
@@ -34,6 +34,91 @@ In der Web Console kann der MySQL (Ephemeral) Service via Catalog dem Projekt hi
...
+
+Schauen wir uns nun die neue Resource Secret mit dem Namen `mysql` an:
```
Comment (bliemli) [inline labs/08_database.md]:
```suggestion
Schauen wir uns nun die neue Ressource Secret mit dem Namen `mysql` an:
```

File: labs/08_database.md [hunk 5]
```diff
@@ -34,6 +34,91 @@ In der Web Console kann der MySQL (Ephemeral) Service via Catalog dem Projekt hi
...
+
+```bash
+$ oc get secret mysql -o yaml
+```
+
+Die entsprechenden Key-Value Pairs sind unter Data ersichtlich:
```
Comment (bliemli) [inline labs/08_database.md]:
```suggestion
Die entsprechenden Key-Value Pairs sind unter `data` ersichtlich:
```

File: labs/08_database.md [hunk 6]
```diff
@@ -34,6 +34,91 @@ In der Web Console kann der MySQL (Ephemeral) Service via Catalog dem Projekt hi
...
+  creationTimestamp: 2018-12-04T10:33:43Z
+  labels:
+    app: mysql-ephemeral
+    template: mysql-ephemeral-template
+  name: mysql
+  ...
+type: Opaque
+```
+
+Die konkreten Werte sind mittels base64 enkodiert. Unter Linux oder in der Gitbash kann man sich den entsprechenden Wert einfach mittels:
```
Comment (bliemli) [inline labs/08_database.md]:
```suggestion
Die konkreten Werte sind mittels base64 enkodiert. Unter Linux oder in der Gitbash kann man sich den entsprechenden Wert einfach mittels:
```
```suggestion
Die konkreten Werte sind base64-kodiert. Unter Linux oder in der Gitbash kann man sich den entsprechenden Wert einfach mittels:
```

File: labs/08_database.md [hunk 7]
```diff
@@ -34,6 +34,91 @@ In der Web Console kann der MySQL (Ephemeral) Service via Catalog dem Projekt hi
...
+
+```bash
+$ echo "YXBwdWlv" | base64 -d
+appuio
+```
+anzeigen lassen. In userem Fall wird `YXBwdWlv` in `appuio` dekodiert.
+
+Mit Secrets können wir also sensitive Informationen(Credetials, Zertifikate, Schlüssel, dockercfg ...) abspeichern und entsprechend von den Pods entkoppeln. Gleichzeitig, haben wir damit die Möglichkeit, die selben Secrets in mehreren Container zu verwenden und so Redundanzen zu vermeiden.
```
Comment (bliemli) [inline labs/08_database.md]:
```suggestion
Mit Secrets können wir also sensitive Informationen (Credetials, Zertifikate, Schlüssel, dockercfg, ...) abspeichern und entsprechend von den Pods entkoppeln. Gleichzeitig haben wir damit die Möglichkeit, dieselben Secrets in mehreren Containern zu verwenden und so Redundanzen zu vermeiden.
```

File: labs/08_database.md [hunk 8]
```diff
@@ -34,6 +34,91 @@ In der Web Console kann der MySQL (Ephemeral) Service via Catalog dem Projekt hi
...
+
+Secrets können entweder wie oben bei der mysql Datenbank in Umgebungsvaribablen gemappt oder direkt als Files via volumes in einen Container gemounted werden.
```
Comment (bliemli) [inline labs/08_database.md]:
```suggestion
Secrets können entweder, wie oben bei der MySQL-Datenbank, in Umgebungsvariablen gemappt oder direkt als Files via Volumes in einen Container gemountet werden.
```

File: labs/08_database.md [hunk 9]
```diff
@@ -34,6 +34,91 @@ In der Web Console kann der MySQL (Ephemeral) Service via Catalog dem Projekt hi
...
+
+Weitere Informationen zu Secrets können hier gefunden werden: https://docs.openshift.com/container-platform/3.9/dev_guide/secrets.html
```
Comment (bliemli) [inline labs/08_database.md]:
```suggestion
Weitere Informationen zu Secrets können in der [offiziellen Dokumentation](https://docs.openshift.com/container-platform/3.9/dev_guide/secrets.html) gefunden werden.
```

File: labs/08_database.md [hunk 10]
```diff
@@ -110,6 +196,50 @@ Die Konfiguration kann auch in der Web Console angeschaut und verändert werden:
 
 (Applications → Deployments → example-spring-boot, Actions, Edit YAML)
 
+## Aufgabe: LAB8.2.1: Setzen der Werte für Usernamen und Passwort aus dem Secret mysql
```
Comment (bliemli) [inline labs/08_database.md]:
```suggestion
## Aufgabe: LAB8.2.1: Secret referenzieren
```

File: labs/08_database.md [hunk 11]
```diff
@@ -110,6 +196,50 @@ Die Konfiguration kann auch in der Web Console angeschaut und verändert werden:
...
+
+Weiter oben haben wir gesehen, wie OpenShift mittels Secrets senisitive Informationen von der eigentlichen Konfiguration enkoppelt und uns dabei hilft Redundanzen zu vermeiden. Unsere Springboot Applikation aus dem vorherigen Lab haben wir zwar korrekt konfiguriert, allerings aber die Werte redundant und plaintext in der DeploymentConfig abgelegt.
```
Comment (bliemli) [inline labs/08_database.md]:
```suggestion
Weiter oben haben wir gesehen, wie OpenShift mittels Secrets sensitive Informationen von der eigentlichen Konfiguration enkoppelt und uns dabei hilft, Redundanzen zu vermeiden. Unsere Springboot Applikation aus dem vorherigen Lab haben wir zwar korrekt konfiguriert, allerings aber die Werte redundant und Plaintext in der DeploymentConfig abgelegt.
```

**Review by bliemli — CHANGES_REQUESTED** [review]

**Comment by phil-pona** [comment]
Thanks for the review and the fixes

**Comment by bliemli** [comment]
Thank you for the PR!

**Pull request merged**